sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2571,15 +2571,12 @@ pi_int32 ExecCGCommand::enqueueImp() {
     std::vector<Requirement *> *CopyReqs =
         new std::vector<Requirement *>(HostTask->getRequirements());
 
-    // Not actually a copy, but move. Should be OK as it's not expected that
-    // MHostKernel will be used elsewhere.
-    std::unique_ptr<HostKernelBase> *CopyHostKernel =
-        new std::unique_ptr<HostKernelBase>(std::move(HostTask->MHostKernel));
+    std::shared_ptr<HostKernelBase> CopyHostKernel = HostTask->MHostKernel;
 
     NDRDescT *CopyNDRDesc = new NDRDescT(HostTask->MNDRDesc);
 
     ArgsBlob[0] = (void *)CopyReqs;
-    ArgsBlob[1] = (void *)CopyHostKernel;
+    ArgsBlob[1] = (void *)CopyHostKernel.get();
```

Comment:
I'm not sure if this means we also need to change this code I've just found for use by  `piEnqueueNativeKernel` https://github.com/intel/llvm/blob/ad372adaf4971b8f543a7aadb2b77994c06d2478/sycl/source/detail/scheduler/commands.cpp#L2296-L2297

Reply:
turned out it was an issue, fixed in https://github.com/reble/llvm/pull/228